trending_data/src/main/webapp/script.js

```diff
@@ -8,16 +8,31 @@ let windowsHandler;
  */
 function initMap() {
   const initPos = new google.maps.LatLng(0, 0);
-  map = new google.maps.Map(document.getElementById('map'), {
+  const map = new google.maps.Map(document.getElementById('map'), {
     center: initPos,
     zoom: 3,
     minZoom: 2,
   });
+
+  getYTSupportedCountries();
   windowsHandler = new UniqueWindowHandler(map);
+
   // Add a marker clusterer to manage the markers.
   const markerCluster = new MarkerClusterer(map, [],
       {imagePath: 'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m'});
   addAllMarkers(markerCluster);
+} /* eslint-enable no-unused-vars */
+
+/**
+ * Fetch country codes that are supported by the
+ * YouTube API and store them in the global array:
+ * ytSupportedCountries
+ */
+function getYTSupportedCountries() {
+  fetch('yt-supported-countries').then((response) =>
+    response.json()).then((supported) => {
+    ytSupportedCountries = supported;
+  });
```

Comment:
Why not put this "supported country check" logic on the backend?

It doesn't matter from a security perspective, since even if a user messes with the request, the worst that will happen is an empty popup on their side.

But from a cost perspective, it makes a difference. Right now, a request is made for every person that enters the app (I mean the 'getsupportedcountries' request). If you put this logic on the backend, you can make this call once when the servlet starts.

This doesn't make much of a difference for us given how restricted the access is, but if the number of users were to grow a lot, this would make a difference.